main.tf

```
@@ -234,3 +234,33 @@ resource "aws_ecr_repository_policy" "name" {
   repository = aws_ecr_repository.name[each.value].name
   policy     = join("", data.aws_iam_policy_document.resource.*.json)
 }
+
+resource "aws_ecr_repository_policy" "ecr_repos_policy" {
+  for_each   = toset(local.ecr_need_policy && module.this.enabled ? local.image_names : [])
```

```suggestion
  for_each   = toset(module.this.enabled ? local.image_names : [])
```
This permission is mandatory for our ECR repos